lib/xenops_server.ml

```diff
@@ -1580,24 +1623,26 @@ and perform_exn ?subtask ?result (op: operation) (t: Xenops_task.task_handle) :
       let dbg = (Xenops_task.to_interface_task t).Task.dbg in
       let url = Uri.of_string url' in
       (* We need to perform version exchange here *)
-      let is_localhost =
-        try
-          let q = query dbg url' in
-          debug "Remote system is: %s" (q |> Query.rpc_of_t |> Jsonrpc.to_string);
-          q.Query.instance_id = instance_id
-        with e ->
-          debug "Failed to contact remote system on %s: is it running? (%s)" url' (Printexc.to_string e);
-          raise (Failed_to_contact_remote_service url') in
-      if is_localhost
-      then debug "This is a localhost migration.";
+
       Xenops_hooks.vm_pre_migrate ~reason:Xenops_hooks.reason__migrate_source ~id;
 
       let module Remote = Xenops_interface.Client(struct let rpc = Xcp_client.xml_http_rpc ~srcstr:"xenops" ~dststr:"dst_xenops" (fun () -> url') end) in
-      let id = Remote.VM.import_metadata dbg (export_metadata vdi_map vif_map vgpu_pci_map id) in
-      debug "Received vm-id = %s" id;
+      let regexp = Re_pcre.regexp id in
+      let newid = (String.sub id 0 24) ^ "000000000001" in
+      let newid2 = (String.sub id 0 24) ^ "000000000000" in
```

Comment:
Calling them `new_dest_id` and `new_orig_id` (or something on that line) is more verbose but also helps not getting confused between `newid` and `newid2` and their uses

Reply:
Yes, good idea. I've confused myself over these things. Especially since newid is 001 and newid2 is 000 !